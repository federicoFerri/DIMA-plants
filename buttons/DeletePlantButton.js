import React from 'react';
import { StyleSheet, Image, TouchableOpacity, SafeAreaView } from 'react-native';

const image_DeletePlantButton = require('../assets/button_images/delete_plant.png');

{/*props required
  @onPress: if button is pressed, it launches an event like TouchableOpacity
*/}

class DeletePlantButton extends React.Component {
    imagePressed = () => () => {
        this.props.onPress()
    }
    render() {
        return(
            <SafeAreaView>
                <TouchableOpacity 
                    activeOpacity={0.5}
                    onPress={this.imagePressed()}
                >
                    <Image 
                        style={stylesImage.deletePlantButton}
                        source={image_DeletePlantButton}
                        /> 
                </TouchableOpacity>
            </SafeAreaView>
        )
    }
  
}


const stylesImage = StyleSheet.create({
    deletePlantButton: {
      width: 18,
      height: 21,
      resizeMode: 'contain',
      //marginTop: 20
    },
  });

export default DeletePlantButton;