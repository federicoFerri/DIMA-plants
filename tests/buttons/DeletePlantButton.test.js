import React from 'react';
import renderer from 'react-test-renderer';

import DeletePlantButton from '../../buttons/DeletePlantButton'

describe('<DeletePlantButton />', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<DeletePlantButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});