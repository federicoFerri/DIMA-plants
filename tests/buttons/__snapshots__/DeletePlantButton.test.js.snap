// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeletePlantButton /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 21,
          "resizeMode": "contain",
          "width": 18,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
